decorator/src/decorator/DressDecorator.java

```diff
@@ -0,0 +1,14 @@
+package decorator;
+
+public class DressDecorator implements Dress{
+    protected Dress dress;
+
+    public DressDecorator(Dress c) {
+        this.dress = c;
+    }
+
+    @Override
+    public void assemble() {
+        this.dress.assemble();
```

Comment:
when this block of code will be executed.
if not it should be an abstract method. or use the default implementation

Reply:
Ya, here we can keep the implementation of assemble method based on the use-case of the dress which we are creating

Reply:
It's like a cherry on the cake. Basic dress has some assemble method/instructions which can be used in any dress which you makes

Reply:
![image](https://user-images.githubusercontent.com/55446681/201330649-6152fb5f-b128-4f51-9eae-f8c825ddf424.png)

here you can see, initially assemble method is called for the basic dress and then the next instructions are printed.

Here is the output:

![image](https://user-images.githubusercontent.com/55446681/201330812-01d8f1c8-5ad1-477f-821d-68f8655124ce.png)